### PR TITLE
feat: add-project-support

### PR DESCRIPTION
**Pull request by chrheg**
# Pull Request
Add Project Support

## Proposed Changes
Adding feature to add issues and pull requests created to a GitHub Project.
This will allow us to track progress over time, and also make it easier to follow up on pending tasks.

I'm not 100% sure about the tests for each of these new functions, as I used Copilot to generate them. 

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by zkoppert** [comment]
What a great idea! This will be helpful in tracking work to completion once the PRs and issues have been opened. Love it! I'll look through the code and give review shortly.

File: evergreen.py [hunk 1]
```diff
@@ -99,6 +115,12 @@ def main():  # pragma: no cover
                         title, body, repo, dependabot_file, commit_message
                     )
                     print("\tCreated pull request " + pull.html_url)
+                    if project_id:
+                        pr_id = get_global_pr_id(
+                            token, organization, repo.name, pull.number
+                        )
+                        link_item_to_project(token, project_id, pr_id)
+                        print("\tLinked pull request to project " + project_id)
```
Comment (zkoppert) [inline evergreen.py:123]: Let's only print success if it was in fact successful:
```suggestion
                        response = link_item_to_project(token, project_id, pr_id)
                        if(response):
                            print("\tLinked pull request to project " + project_id)
```

Comment (chrheg) [reply evergreen.py]: Good catch , I agree that's a lot better. The ID that get's printed is the Global ID so does not give to much value.

**Review by zkoppert — APPROVED** [review]
Fantastic! 🚀

File: evergreen.py [hunk 2]
```diff
@@ -99,6 +115,13 @@ def main():  # pragma: no cover
                         title, body, repo, dependabot_file, commit_message
                     )
                     print("\tCreated pull request " + pull.html_url)
+                    if project_id:
+                        pr_id = get_global_pr_id(
+                            token, organization, repo.name, pull.number
+                        )
+                        response = link_item_to_project(token, project_id, pr_id)
+                        if(response):
```
Comment (zkoppert) [inline evergreen.py:123]:
```suggestion
                        if response:
```

**Pull request merged**